src/compiler/checker.ts

```diff
@@ -27570,6 +27572,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (assignmentKind === AssignmentKind.Definite) {
                 return type;
             }
+            if (assignmentKind === AssignmentKind.CompoundLike) {
+                return getBaseTypeOfLiteralType(type);
+            }

```

Comment:
That sounds okay to me, but, I also don't know how much effort or how big of a change that ends up being either.

Reply:
Just going back over unreviewed PRs; where are you at with the above? (All good if nowhere, but, I probably want to wait for the above for this PR.)

Reply:
I didn't have time to work on this one (and it's not the priority one for me so it might take a while before I get to it). My memory is a little bit fuzzy on this one... I **think** that I would like to propose a change that would make those 2 markers display the same type:
```ts
let x: SomeType
x/*1*/ = expr
x/*2*/
```

I think it makes sense and it would make it easier to reason about some of the baselines here. That is a separate change, so I will open a new issue about it and discuss it there. If that would be accepted and implemented (happy to do that) then we could circle back to this PR here and focus on the problem at hand. How does it sound?

> I probably want to wait for the above for this PR

Did you mean what I just described by "the above" here?

Reply:
> That is a separate change, so I will open a new issue about it and discuss it there. If that would be accepted and implemented (happy to do that) then we could circle back to this PR here and focus on the problem at hand. How does it sound?

Yes, I feel like that makes sense, especially in context of this hack.

> Did you mean what I just described by "the above" here?

Yes, in that I like fewer hacks 😄